### PR TITLE
selinux: Initial commit for clear-containers selinux modules

### PR DESCRIPTION
**Pull request by gorozco1**
selinux: Initial commit for cc-proxy selinux module
Initial commit to allow cc-proxy to run with selinux enforced.
This branch fixes issue #519 "systemd cc-proxy.socket cannot
be started in Fedora 24 when selinux=enforced"

Signed-off-by: Geronimo Orozco <geronimo.orozco@intel.com>

### CODE REVIEW

**Comment by chavafg** [comment]
qa-failed

**Comment by dlespiau** [comment]
Could you put the file in the proxy/ directory and the amend the existing README there?

Also, more than the links on the bug, I'd rather have the explanation on how to use the policy.

**Comment by sameo** [comment]
Adding @ereshetova for review.
Elena, this is supposed to fix #519, and we'd appreciate if you could review the te file.

**Comment by sameo** [comment]
@gorozco1 Please also specify which issue this is fixing, in your commit message.

**Comment by ereshetova** [comment]
There is no reason not to combine permissions:

+allow init_t unconfined_service_t:unix_stream_socket listen;
+
+#!!!! This avc is allowed in the current policy
+allow init_t unconfined_service_t:unix_stream_socket { bind create setopt };

Should become:

+allow init_t unconfined_service_t:unix_stream_socket { bind create setopt listen };

I think base selinux doesn't have macros, so just calling out exact access types are ok. 

One think I would double check in your case is that the process running in the unconfined_service_t domain that needs new allow to be added is actually supposed to run in that domain. Just to double check that no mistakes were done.

**Comment by gorozco1** [comment]
@ereshetova updated module thanks for your feedback.

**Comment by gorozco1** [comment]
@dlespiau updated module, path and readmes

**Comment by sameo** [comment]
@gorozco1 2 comments:
1. Can we move the selinux directory under `proxy/` ? We're going to move the proxy code into its own repo, so better separate the selinux specific bits now.
2. I'm fine with carrying the spec and deb files, but I'd like it to be in a dedicated directory where we gather all meta file for each distro.

**Comment by gorozco1** [comment]
@sameo done, moved to proxy and removed the spec file.

File: proxy/README.md [hunk 1]
```diff
@@ -113,6 +113,42 @@ handled by systemd and can be viewed with:
 journalctl -u cc-proxy -f
 ```
 
+## SELinux
+
+`cc-proxy` can't run out of the box on fedora systems running SELinux enforced. 
+This is documented in issue #519
```
Comment (dlespiau) [inline proxy/README.md]: Can you remove this sentence? the necessary information should be part of this README, the "how we got there" does't really matter to the end user.

**Comment by sameo** [comment]
LGTM

**Comment by jcvenegas** [comment]
lgtm

**Pull request merged**